src/main/java/org/jqassistant/contrib/plugin/plantumlrule/CucaDiagramParser.java

```diff
@@ -107,20 +112,23 @@ private void addNode(IEntity entity, Map<String, Node> nodes) {
             if (relationshipLabel != null) {
                 Relationship.RelationshipBuilder builder = Relationship.builder().id(link.getUid().toLowerCase());
                 builder.relationshipLabel(relationshipLabel);
-                String relationType = relationshipLabel.getType();
                 Node entity1 = nodes.get(link.getEntity1().getUid());
                 Node entity2 = nodes.get(link.getEntity2().getUid());
-                if (link.isInverted()) {
+                if (entity1 != null && entity2 != null) {
+                  if (link.isInverted()) {
                     builder.from(entity2);
                     builder.to(entity1);
+                  } else {
+                      builder.from(entity1);
+                      builder.to(entity2);
+                  }
+                  Relationship relationship = builder.build();
+                  relationships.put(relationship.getId(), relationship);
                 } else {
-                    builder.from(entity1);
-                    builder.to(entity2);
+                  LOGGER.info("Ignoring relationship '{}' because of node without stereotype.", display);
```

Comment:
Do we need this in info level? Looks like a ususal situation where no extra logging is required. So could be debug.

Reply:
You are right. I changed it.